storage/timestamp_cache.go

```diff
@@ -36,6 +36,24 @@ const (
 	MinTSCacheWindow = 10 * time.Second
 )
 
+var (
+	// evictionSizeThreshold is the maximum allowed number of intervals
+	// in the TimestampCache before it will begin evicting intervals
+	// based on the time window. This threshold is intended to permit
+	// transactions to take longer than the evition window duration
+	// if the size of the cache is not a concern, such as when a user
+	// is using an interactive SQL shell.
+	evictionSizeThreshold = 512
```

Comment:
do we really need to throw around globals for this? Just make it a setting on the timestamp cache.